src/hotspot/share/memory/universe.cpp

```diff
@@ -89,6 +89,9 @@ Klass* Universe::_typeArrayKlassObjs[T_LONG+1]        = { NULL /*, NULL...*/ };
 Klass* Universe::_objectArrayKlassObj                 = NULL;
 Klass* Universe::_fillerArrayKlassObj                 = NULL;
 OopHandle Universe::_mirrors[T_VOID+1];
+#if INCLUDE_CDS_JAVA_HEAP
+int Universe::_archived_mirror_indices[T_VOID+1];
```

Comment:
a minor suggestion - I feel it would be better to name it as `_archived_basic_type_mirror_indices` as it only stores mirrors for primitive types. Same for the other functions accessing `_archived_mirror_indices` i.e. `set_archived_mirror_index` -> `set_archived_basic_type_mirror_index` and `update_archived_mirrors` -> `update_archived_basic_type_mirrors`.

Reply:
I renamed the field/functions as you suggested. Now they are consistent with existing functions like `Universe::initialize_basic_type_mirrors()`

I think eventually they should be named to `primitive_type_mirrors` since we don't use them for the reference types. But there are a few functions that are unrelated to this RFE, so I'll probably do the renaming in a subsequent RFE.